src/mapsharing.cpp

```diff
@@ -105,6 +105,20 @@ void MAP_SHARING::setDefaults()
 
 void ofstream_wrapper::open( const std::ios::openmode mode )
 {
+#if defined(_WIN32)
```

Comment:
I think this can simply call the `fs::path` override of `as_norm_dir`.

Reply:
Replaced with `is_lexically_valid` and throws if not.